README.md

```diff
@@ -1,2 +1,196 @@
 # Zod to OpenAPI
 
+A library that uses zod schemas to generate an Open API Swagger documentation.
+
+1. [Purpose](#purpose)
+2. [How it works](#how-it-works)
+   1. [Technologies](#technologies)
+   2. [Concept](#concept)
+3. [Quick start](#quick-start) — Fast Track
+   1. [Installation](#installation)
+   2. [Set up](#set-up)
+
+<!-- TODO: Something about a CHANGELOG -->
+
+## Purpose
+
+We at Astea Solutions made this library because of the duplication of work when creating an API validated API with `zod` that needs documentation.
+
+For example:
+
+We can have a zod schema like
+
+```ts
+const UserIdSchema = z.object({
+  id: z.string(),
+});
+const UserSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  age: z.number(),
+});
+```
+
+this schema can be used to validate the response of a `GET /users/{id}` endpoint. So the documentation may look like
+
+```yaml
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: string
+          example: '1212121'
+        name:
+          type: string
+          example: 'John Doe'
+        age:
+          type: number
+          example: 42
+      required:
+        - userId
+        - name
+        - age
+  parameters:
+    UserId:
+      in: header
+      name: id
+      schema:
+        type: string
+      required: true
+      description: The user id to look for
+      allowReserved: true
+      example: '1212121'
+# ... Some other stuff in between
+paths:
+  /users/{id}:
+    get:
+      description: Get user data by its id
+      summary: Get a single user
+      parameters:
+        - $ref: '#/components/parameters/UserId'
+      responses:
+        '200':
+          description: Object with user data.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+```
+
+Obviously there is a lot of duplication when defining the zod schema and the schema for Open API. And let's face it - writing code with intellisense is much easier than writing YAML.
+
+### The solution
+
+With z
+
+```ts
+extendZodWithOpenApi(z);
```

Comment:
Please show the imports as well and mention that this should be done only once in a common-entrypoint file.